api/adscert.proto

```diff
@@ -38,40 +44,49 @@ message AuthenticatedConnectionSignatureRequest {
 // request, including any signature and relevant metadata. Multiple signatures
 // can technically be present according to the specification.
 message AuthenticatedConnectionSignatureResponse {
-    SignatureStatus signature_status = 1;
+    SignatureOperationStatus signature_operation_status = 1;
     RequestInfo request_info = 2;
 }
 
-enum SignatureStatus {
-    SIGNATURE_STATUS_UNDEFINED = 0;
-    SIGNATURE_STATUS_OK = 1;
-    SIGNATURE_STATUS_SIGNATORY_DEACTIVATED = 2;
-    SIGNATURE_STATUS_SIGNATORY_INTERNAL_ERROR = 3;
-    SIGNATURE_STATUS_MISSING_REQUIRED_PARAMETER = 4;
-    SIGNATURE_STATUS_NO_COUNTERPARTY_INFO = 5;
-}
-
 // AuthenticatedConnectionVerificationRequest contains a request for verifying
 // signatures generated by another party.
 message AuthenticatedConnectionVerificationRequest {
-    RequestInfo request_info = 1;
+    repeated RequestInfo request_info = 1;
 }
 
 // AuthenticatedConnectionVerificationResponse contains the results of verifying
 // signatures.
 message AuthenticatedConnectionVerificationResponse {
-    VerificationStatus verification_status = 1;
-    bool body_valid = 2;
-    bool url_valid = 3;
+    VerificationOperationStatus verification_operation_status = 1;
+    repeated RequestVerificationInfo verification_info = 2;
+}
+
+enum SignatureDecodeStatus {
+    SIGNATURE_DECODE_STATUS_UNDEFINED = 0;
+    SIGNATURE_DECODE_STATUS_BODY_AND_URL_VALID = 1;
+    SIGNATURE_DECODE_STATUS_BODY_VALID = 2;
+    SIGNATURE_DECODE_STATUS_INVALID_SIGNATURE = 3;
+    SIGNATURE_DECODE_STATUS_SIGNATURE_NOT_PRESENT = 4;
+    SIGNATURE_DECODE_STATUS_SIGNATURE_MALFORMED = 5;
+    SIGNATURE_DECODE_STATUS_UNRELATED_SIGNATURE = 6;
+    SIGNATURE_DECODE_STATUS_COUNTERPARTY_LOOKUP_ERROR = 7;
+    SIGNATURE_DECODE_STATUS_NO_SHARED_SECRET_AVAILABLE = 8;
+}
+
+enum SignatureOperationStatus {
+    SIGNATURE_OPERATION_STATUS_UNDEFINED = 0;
+    SIGNATURE_OPERATION_STATUS_OK = 1;
+    SIGNATURE_OPERATION_STATUS_SIGNATORY_DEACTIVATED = 2;
+    SIGNATURE_OPERATION_STATUS_SIGNATORY_INTERNAL_ERROR = 3;
+    SIGNATURE_OPERATION_STATUS_MALFORMED_REQUEST = 4;
 }
 
-enum VerificationStatus {
-    VERIFICATION_STATUS_UNDEFINED = 0;
-    VERIFICATION_STATUS_OK = 1;
-    VERIFICATION_STATUS_SIGNATORY_DEACTIVATED = 2;
-    VERIFICATION_STATUS_SIGNATORY_INTERNAL_ERROR = 3;
-    VERIFICATION_STATUS_MISSING_REQUIRED_PARAMETER = 4;
-    VERIFICATION_STATUS_NO_COUNTERPARTY_INFO = 5;
```

Comment:
do we not want this one anymore?

Reply:
What part are you referring to? 

`VerificationStatus` is now `VerificationOperationStatus` referring to the actual function call to the signatory to verify signatures for a request. The `NO_COUNTERPARTY_INFO` option was removed because that's not related to the function call itself (which is really whether the signatory did the processing or threw an error for some reason).

The actual decode status for an individual signature is a separate new `SignatureDecodeStatus` returned in the `RequestVerificationInfo` message type.

Reply:
I concur with this change